### PR TITLE
Add environment hook for CLASSPATH

### PR DESCRIPTION
**Pull request by jacobperron**
This is a quick fix to a problem described in https://github.com/esteve/ament_java/issues/9

TL;DR
Jars produced by gradle packages are not added to the CLASSPATH, and so this PR let's colcon handle it.

Maybe this is better suited to go in colcon-ros-gradle, I don't know. But, long-term, I think we should have a standalone Java library that contains helpers for registering ament packages. Such a library can be used by the ament gradle plugin or elsewhere.

### CODE REVIEW

**Comment by jpace121** [comment]
How likely is this to get merged?

I'm trying to use colcon to build a non-ros mixed language workspace, including some components built with gradle, and this PR was very helpful in getting that working.

**Comment by esteve** [comment]
@jacobperron I agree this should be in `colcon-ros-gradle`, how hard would it be to port it there?

**Comment by ivanpauno** [comment]
> @jacobperron thanks! I agree this should be in `colcon-ros-gradle`, how hard would it be to port it there?

IMHO this patch seems well suited to here, e.g. "colcon python" adds modules to `PYTHONPATH` in an environment hoook, and not "colcon ros python" (see [here](https://github.com/colcon/colcon-core/blob/db0535c8056238131db5cd273c7a52fe5a2cd68f/colcon_core/task/python/build.py#L139-L141)).

Anyways, getting this in any of the two works for me :smile: .

File: colcon_gradle/task/gradle/build.py [hunk 1]
```diff
@@ -56,14 +58,25 @@ def add_arguments(self, *, parser):  # noqa: D102
...
         logger.info(
             "Building Gradle package in '{args.path}'".format_map(locals()))
 
+        # add jars and classes to CLASSPATH with wildcards
+        # https://docs.oracle.com/javase/8/docs/technotes/tools/windows/classpath.html#A1100762
+        additional_hooks += create_environment_hook(
+            'classpath_jars', Path(args.install_base), pkg.name,
+            'CLASSPATH', os.path.join('share', pkg.name, 'java', '*'),
+            mode='prepend')
+        additional_hooks += create_environment_hook(
```
Comment (ivanpauno) [inline colcon_gradle/task/gradle/build.py:75]: do we also need the directory?
I think that the wildcard is enough

Comment (jacobperron) [reply colcon_gradle/task/gradle/build.py:75]: Reading the the [docs](https://docs.oracle.com/javase/8/docs/technotes/tools/windows/classpath.html#A1100762), I'm lead to believe we need both the path and the wildcard to account for class files AND jars:

> A class path entry that contains an asterisk (*) does not match class files. To match both classes and JAR files in a single directory mydir, use either mydir:mydir/* or mydir/*:mydir. The order chosen determines whether the classes and resources in mydir are loaded before JAR files in mydir or vice versa.

File: colcon_gradle/task/gradle/build.py [hunk 2]
```diff
@@ -56,14 +58,25 @@ def add_arguments(self, *, parser):  # noqa: D102
             help='Run a specific task instead of the default task')
 
     async def build(
-        self, *, additional_hooks=None, skip_hook_creation=False
+        self, *, additional_hooks=[], skip_hook_creation=False
```
[the remark this answers is not in the document]
Comment (jacobperron) [reply colcon_gradle/task/gradle/build.py]: Good catch! I've rebased my changes on `master` and fixed this. PTAL

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**